lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -4016,6 +4017,13 @@ void ByteCodeGenerator::StartEmitFunction(ParseNode *pnodeFnc)
                 this->EnsureLetConstScopeSlots(pnodeFnc->sxFnc.pnodeBodyScope, funcInfo);
             }
         }
+
+        if (!paramScope->GetCanMergeWithBodyScope() && bodyScope->GetScopeSlotCount() == 0 && !bodyScope->GetHasOwnLocalInClosure())
+        {
+            // When we have split scope the body scope may be wrongly marked as must instantiate even though the capture occurred
+            // in param scope. This check is to make sure if no capture occurs in body scope make in not must instantiate.
+            bodyScope->SetMustInstantiate(false);
```

Comment:
Then what is the point on doing it on line 3807? why not move that here?

---
In reply to: [61648793](https://github.com/Microsoft/ChakraCore/pull/892#discussion_r61648793) [](ancestors = 61648793)